src/reconcile/rec_visibility.c

```diff
@@ -254,17 +258,20 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         if (upd->prepare_state == WT_PREPARE_LOCKED ||
           upd->prepare_state == WT_PREPARE_INPROGRESS) {
             WT_ASSERT(session, upd_select->upd == NULL);
-            has_newer_updates = true;
-            if (upd->start_ts > max_ts)
-                max_ts = upd->start_ts;
-
-            /*
-             * Track the oldest update not on the page, used to decide whether reads can use the
-             * page image, hence using the start rather than the durable timestamp.
-             */
-            if (upd->start_ts < r->min_skipped_ts)
-                r->min_skipped_ts = upd->start_ts;
-            continue;
+            /* Ignore prepared updates if it is not eviction. */
+            if (!F_ISSET(r, WT_REC_EVICT)) {
```

Comment:
Can we merge this test with the outer if statement? If you are worried about the assert, probably we should remove that.

Reply:
Fixed.